src/nvim/eval.c

```diff
@@ -8141,19 +8144,23 @@ static void f_execute(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     }
     if (strncmp(s, "silent", 6) == 0) {
       msg_silent++;
+      is_silent = true;
     }
     if (strcmp(s, "silent!") == 0) {
       emsg_silent = true;
       emsg_noredir = true;
     }
   } else {
     msg_silent++;
+    is_silent = true;
   }
 
   garray_T capture_local;
   ga_init(&capture_local, (int)sizeof(char), 80);
   capture_ga = &capture_local;
   redir_off = false;
+  const bool save_did_emsg = did_emsg;
```

Comment:
It's customary to keep the `save_xx` declarations grouped together (line 8126) if possible.